src/bokeh/palettes.py

```diff
@@ -1639,6 +1639,43 @@ def set_alpha(rgba: RGB, i: int) -> RGB:
 
     return palette
 
+def interp_palette(palette: Palette, n: int) -> Palette:
+    """ Generate a new palette by interpolating a given palette.
+
+    Linear interpolation is performed separately on each of the RGBA
+    components.
+
+    Args:
+        palette (seq[str]) :
+            A sequence of hex RGB(A) color strings to create new palette from
+
+        n (int) :
+            The size of the palette to generate
+
+    Returns:
+        seq[str] : a sequence of hex RGB(A) color strings
+
+    Raises:
+        ValueError if n is negative
```

Comment:
My experience with other python codebases is only to list explicitly raised exceptions

Reply:
I'll add the other explicit cause of `ValueError` here.

For the wider issue of if this is the best place for exceptions to be mentioned, so far I have just followed the same format as our other Python docstrings which do it this way. I can see the logic of mentioning `ValueError`s nearer the argument they refer to. Presumably we would still potentially need a separate `Raises` section for exceptions that aren't just associated with a specific argument. But then if we have a `Raises` section shouldn't all exceptions be documented in it? Anyway, if that is what we want to do it should be standardised across the codebase rather than start in this PR.